### PR TITLE
Persist provisioned accounts

### PR DESCRIPTION
**Pull request by ChristophWurst**
*Default accounts* will be a bit tricky to handle in the future, as we will often need access to accounts, even if the user is not logged in (background jobs). Moreover, you can't persist any settings as they all share the same -2 id across all users.

Thus, the idea is to just persist them. Admins shall from now on configure this only through the web UI, hence adding a settings interface.

Todo

- [x] Add admin settings page
- [x] Add admin provisioning settings
  - [x] Basic settings form
  - [x] Preview for easier template creation
  - [x] Persist settings
  - [x] Update all existing provisioned accounts and create new for all missing
- [x] Provision new account when user is created
- [x] When a user accesses a provisioned account, set the (until then) missing password
- [x] Add repair step to also create these accounts from the old manual config
- [x] Prevent deletion of the default account

---

Current (2019-11-15) state

Bildschirmfoto von 2019-11-15 14-31-16

### CODE REVIEW

File: lib/Migration/MigrateProvisioningConfig.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types=1);
+
+/**
+ * @author Christoph Wurst <christoph@winzerhof-wurst.at>
```
Comment (rullzer) [inline lib/Migration/MigrateProvisioningConfig.php]: whoops

Comment (ChristophWurst) [reply lib/Migration/MigrateProvisioningConfig.php]: dang it, phpstorm!

**Comment by andreas-p** [comment]
Came across this from closing #425 info: I wonder if I under stand the check "When a user accesses a provisioned account, set the (until then) missing password" right: The password for the (formerly called) default account will be stored in the database, just as it is done for quite a while with additional mail accounts?

**Comment by ChristophWurst** [comment]
Yes, that is correct. For further questions please use https://help.nextcloud.com/c/apps/mail :)

**Pull request merged**